packages/typespec-client-generator-core/test/packages/parameters.test.ts

```diff
@@ -0,0 +1,1005 @@
+import { AzureCoreTestLibrary } from "@azure-tools/typespec-azure-core/testing";
+import { deepStrictEqual, ok, strictEqual } from "assert";
+import { beforeEach, describe, it } from "vitest";
+import {
+  SdkHeaderParameter,
+  SdkHttpOperation,
+  SdkPathParameter,
+  SdkQueryParameter,
+  SdkServiceMethod,
+} from "../../src/interfaces.js";
+import { SdkTestRunner, createSdkTestRunner } from "../test-host.js";
+import { getServiceMethodOfClient, getServiceWithDefaultApiVersion } from "./utils.js";
+
+describe("typespec-client-generator-core: parameters", () => {
+  let runner: SdkTestRunner;
```

Comment:
The test files under `packages` folder are all used to test `sdkpackage`? Shall we move `package.test.ts` under this `package` folder as well?

Reply:
yes. i'll do this step by step. i don't want to add too many refactor code in this pr, just move the parameter's tests first.